README.md
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->


# Custom Network Image

```custom_network_image```  A Flutter library to show images from the internet with custom design and decoration.  ✨

It's support some type of image decoration such as [ sizeOfImage, isCircular, isCircularBorder, isCurve or isCurveBorder], etc.


## Installation

Create a new project with the command

```yaml
flutter create MyApp
```

Add

```yaml
custom_network_image: ...
```

to your `pubspec.yaml` of your flutter project.
**OR**
run

```yaml
flutter pub add custom_network_image
```

in your project's root directory.

In your library add the following import:

```dart
import 'package:custom_network_image/custom_network_image.dart';
```

For help getting started with Flutter, view the online [documentation](https://flutter.io/).

## Usage

```dart
class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Custom Network Image',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          backgroundColor: Theme.of(context).colorScheme.inversePrimary,
          title: Text(widget.title),
        ),
        body: Center(
          child: Column(children: [
            const SizedBox(height: 10,),
            CustomNetworkImage(sizeOfImage: 65, url: 'https://picsum.photos/250?image=9', isCircular: true,),
            const SizedBox(height: 10,),
            CustomNetworkImage(curveValue: 10 ,sizeOfImage:  45, url: 'https://picsum.photos/250?image=9', isCurve: true,),
            const SizedBox(height: 10,),
            CustomNetworkImage(borderColor : Colors.purple , radius : 20 ,url :'https://picsum.photos/250?image=9' ,sizeFromRadius:  48,borderThickness:   3 , isCurveBorder : true),
            const SizedBox(height: 10,),
            CustomNetworkImage(borderColor : Colors.purple , radius : 20 ,url :'https://picsum.photos/250?image=9' ,sizeFromRadius:  48,borderThickness:   3 , isCircularBorder : true),
          ],),
        )
    );
  }
}
```


## Constructor

#### Basic

| Parameter       | Default | Description                          | Required |
|-----------------|:--------|:-------------------------------------|:--------:|
| url             | -       | Image base url.                      |   True   |
| curveValue      | -       | Curve value if curve is set to true. |  false   |
| borderColor     | -       | Image border color.                  |  false   |
| radius          | -       | Circle avatar radius.                |  false   |
| borderThickness | -       | Image border thickness               |  false   |
| isCircular      | false   | Set true if show circular box        |  false   |
| isCurve         | false   | Image curve flag                     |  false   |
| isCurveBorder   | false   | Image curve wit border flag          |  false   |


CHANGELOG.md
## 0.0.1

- 0.0.1: initial release of custom network image package..
- 0.0.2: Document format and bug fixes..
- 0.0.3: fixes for code has no errors, warnings, lints, or formatting issues..

test/custom_network_image_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:custom_network_image/custom_network_image.dart';

void main() {

}


lib/custom_network_image.dart
/// Documentation
///
/// custom_network_image library collection.
library custom_network_image;

export 'circular_network_image.dart';
import 'package:custom_network_image/circular_network_image.dart';
import 'package:custom_network_image/curve_network_image.dart';
import 'package:flutter/material.dart';
import 'circular_network_image_with_border.dart';
import 'curve_network_border_image.dart';

///
/// CustomNetworkImage main class..
class CustomNetworkImage extends StatelessWidget {
  const CustomNetworkImage({
    super.key,
    this.sizeOfImage,
    required this.url,
    this.curveValue,
    this.borderColor,
    this.sizeFromRadius,
    this.borderThickness,
    this.radius,
    this.isCircular = false,
    this.isCircularBorder = false,
    this.isCurve = false,
    this.isCurveBorder = false,
  });

  final double? sizeOfImage;
  final String? url;
  final double? curveValue;
  final Color? borderColor;
  final double? sizeFromRadius;
  final double? borderThickness;
  final double? radius;
  final bool? isCircular;
  final bool? isCircularBorder;
  final bool? isCurve;
  final bool? isCurveBorder;

  @override
  Widget build(BuildContext context) {
    return isCircular == true
        ? CircularNetworkImage(
            sizeOfImage: sizeOfImage,
            url: url,
          )
        : isCurve == true
            ? CurveNetworkImage(
                sizeOfImage: sizeOfImage,
                url: url,
                curveValue: curveValue,
              )
            : isCircularBorder == true
                ? CircularImageWithBorder(
                    borderColor: borderColor,
                    sizeFromRadius: sizeFromRadius,
                    url: url,
                    padding: borderThickness)
                : CurveNetworkBorderImage(
                    borderColor: borderColor,
                    sizeFromRadius: sizeFromRadius,
                    url: url,
                    padding: borderThickness,
                    radius: radius,
                  );
  }
}


lib/curve_network_border_image.dart
import 'package:flutter/material.dart';

///
/// CurveNetworkBorderImage StatelessWidget Widget..
class CurveNetworkBorderImage extends StatelessWidget {
  const CurveNetworkBorderImage({
    super.key,
    this.borderColor,
    this.sizeFromRadius,
    this.url,
    this.padding,
    this.radius,
  });

  final Color? borderColor;
  final double? sizeFromRadius;
  final String? url;
  final double? padding;
  final double? radius;

  @override
  Widget build(BuildContext context) {
    final borderRadius = BorderRadius.circular(radius ?? 20); // Image border
    return Container(
      padding: EdgeInsets.all(padding ?? 3), // Border width
      decoration: BoxDecoration(
          color: borderColor ?? Colors.red, borderRadius: borderRadius),
      child: ClipRRect(
        borderRadius: borderRadius,
        child: SizedBox.fromSize(
          size: Size.fromRadius(sizeFromRadius ?? 48), // Image radius
          child: Image.network(url ?? 'https://picsum.photos/250?image=9',
              fit: BoxFit.cover),
        ),
      ),
    );
  }
}


lib/circular_network_image_with_border.dart
import 'package:flutter/material.dart';

///
/// CircularImageWithBorder StatelessWidget Widget..
class CircularImageWithBorder extends StatelessWidget {
  const CircularImageWithBorder(
      {super.key,
      this.borderColor,
      this.sizeFromRadius,
      this.url,
      this.padding});

  final Color? borderColor;
  final double? sizeFromRadius;
  final String? url;
  final double? padding;
  @override
  Widget build(BuildContext context) {
    return Container(
      padding: EdgeInsets.all(padding ?? 3), // Border width
      decoration: BoxDecoration(color: borderColor, shape: BoxShape.circle),
      child: ClipOval(
        child: SizedBox.fromSize(
          size: Size.fromRadius(sizeFromRadius ?? 48), // Image radius
          child: Image.network(url ?? 'https://picsum.photos/250?image=9',
              fit: BoxFit.cover),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:custom_network_image/custom_network_image.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: const Text("Custom Network Image"),
      ),
      body: const Center(
        child: Column(children: [
          SizedBox(height: 10,),
          CustomNetworkImage(sizeOfImage: 65, url: 'https://picsum.photos/250?image=9', isCircular: true,),
          SizedBox(height: 10,),
          CustomNetworkImage(curveValue: 10 ,sizeOfImage:  45, url: 'https://picsum.photos/250?image=9', isCurve: true,),
          SizedBox(height: 10,),
          CustomNetworkImage(borderColor : Colors.purple , radius : 20 ,url :'https://picsum.photos/250?image=9' ,sizeFromRadius:  48,borderThickness:   3 , isCurveBorder : true),
          SizedBox(height: 10,),
          CustomNetworkImage(borderColor : Colors.purple , radius : 20 ,url :'https://picsum.photos/250?image=9' ,sizeFromRadius:  48,borderThickness:   3 , isCircularBorder : true),

        ],),
      )
    );
  }
}
